### PR TITLE
Fix native macOS build for ALL_BUILD (Xcode/AppleClang)

### PR DESCRIPTION
**Pull request by traceon**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Not for changelog

Changelog entry:
- None.

Detailed description:
- Removed unused local variable definition in `CacheDictionaryStorage` member function.
- Not building `keeper-data-dumper` util if `USE_NURAFT` does not evaluate to `True`.
- Fixed the way C++20 standard flags are set (based on the CMake version). This may impact other builds too, but should do that in a positive way.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -248,19 +248,27 @@ if (ARCH_NATIVE)
...
+    set (CMAKE_CXX_STANDARD 20)
+    set (CMAKE_CXX_EXTENSIONS ON) # Same as gnu++2a (ON) vs c++2a (OFF): https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html
+    set (CMAKE_CXX_STANDARD_REQUIRED ON)
+endif ()
 
-set (CMAKE_CXX_EXTENSIONS 0) # https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html#prop_tgt:CXX_EXTENSIONS
-set (CMAKE_CXX_STANDARD_REQUIRED ON)
+#set (CMAKE_C_STANDARD 11)
+#set (CMAKE_C_EXTENSIONS ON)
+#set (CMAKE_C_STANDARD_REQUIRED ON)
```
Comment (traceon) [inline CMakeLists.txt:271]: I suggest setting these too, for consistency:
```suggestion
set (CMAKE_C_STANDARD 11)
set (CMAKE_C_EXTENSIONS ON)
set (CMAKE_C_STANDARD_REQUIRED ON)
```

Comment (alexey-milovidov) [reply CMakeLists.txt:271]: Ok.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> test_materialize_mysql_database

Bad test - @zhang2014 @BohuTANG

**Comment by alexey-milovidov** [comment]
@traceon "Build/Testing/Packaging Improvement" cannot have no changelog. I will change the category to "Not for changelog".

File: utils/CMakeLists.txt [hunk 2]
```diff
@@ -34,6 +34,10 @@ if (NOT DEFINED ENABLE_UTILS OR ENABLE_UTILS)
     add_subdirectory (wal-dump)
     add_subdirectory (check-mysql-binlog)
 
+    if (USE_NURAFT)
+        add_subdirectory (nukeeper-data-dumper)
+    endif ()
+
```
Comment (traceon) [inline utils/CMakeLists.txt:40]:
```suggestion
```

**Comment by alexey-milovidov** [comment]
keeper-data-dumper is the same as nukeeper-data-dumper, so it should be under USE_NURAFT.

**Comment by traceon** [comment]
@alexey-milovidov yeah, noticed. Re-adding.

**Pull request merged**